spec/tracking_system_spec.rb

```diff
@@ -0,0 +1,252 @@
+      end
+      expect{@tracker.view_available_rooms_on(start_time: Date.new(2018,1,1),end_time:Date.new(2018,1,2))}.must_raise ArgumentError
+    end
+  end
+
+
+  describe "#add_reservation" do
+    before do
+      @tracker = TrackingSystem.new
+    end
+
+    it "returns an array of reservations" do
+      @reservations = @tracker.add_reservation(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:1)
+      @reservations = @tracker.add_reservation(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:6)
+      list = @tracker.reservations
+      expect(@reservations).must_be_kind_of Array
+    end
+
+    it "raises ArgumentError if there are not enough available-rooms on the requested date" do
+      expect{@tracker.add_reservation(start_time: Date.new(2018,1,1),end_time:Date.new(2018,1,2),number_of_rooms:350)}.must_raise ArgumentError
+    end
+
+    it "increases the number of reservations in the reservations list" do
+      @reservations = @tracker.add_reservation(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:1)
+      original_num_of_reservations = @tracker.reservations
+      @reservations = @tracker.add_reservation(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:1)
+      updated_num_of_reservations = @tracker.reservations
+      expect(updated_num_of_reservations.length).must_equal 2
+    end
+  end
+
+
+
+  describe "#add_block" do
+    before do
+      @tracker = TrackingSystem.new
+    end
+#
+    it "raises ArgumentError if discount-rate is not an Integer" do
+      expect{@tracker.add_block(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,5), rooms: 1, discount: "hi")}.must_raise ArgumentError
+    end
+
+    it "raises ArgumentError if rooms > 5 or <= 1" do
+      @tracker = TrackingSystem.new
+      expect{@tracker.add_block(start_time: Date.new(2018,8,5),end_time:Date.new(2018,9,5), rooms: 20, discount: 10)}.must_raise ArgumentError
+    end
+
+    it "returns an array of blocks" do
+      @block = @tracker.add_block(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:5)
+      @block = @tracker.add_block(start_time: Date.new(2018,7,1), end_time: Date.new(2018,7,10), number_of_rooms:4)
+      expect(@tracker.blocks).must_be_kind_of Array
+    end
+
+    it "raises ArgumentError if start_time is > end_time" do
+      @tracker = TrackingSystem.new
+      expect{@tracker.add_block(start_time: Date.new(2018,10,5),end_time:Date.new(2018,9,5))}.must_raise ArgumentError
+    end
+
+    it "modifys each room's block id to the same block id as its Block" do
+      @block = @tracker.add_block(start_time: Date.new(2018,8,1), end_time: Date.new(2018,8,25), number_of_rooms:5)
+      @blockid = @tracker.blocks[0].block
+      @room_block = @tracker.blocks[0].rooms[0].block
+      expect(@blockid == @room_block).must_equal true
+    end
+  end
+
+  describe "#retrieve_block_dates" do
```

Comment:
I would suggest using a block Id number instead of a symbol just because it's easier to work with.